cpptests/t_basic.cpp

```diff
@@ -31,4 +69,215 @@ TEST_F(FormatterTest, testFormatCommands) {
     expected = "*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$0\r\n\r\n";
     ASSERT_EQ(expected, formatCommand("SET %s %s", "foo", ""))
         << "empty string";
+
+    expected = "*3\r\n$3\r\nSET\r\n$0\r\n\r\n$3\r\nfoo\r\n";
+    ASSERT_EQ(expected, formatCommand("SET %s %s","","foo"))
+        << "an empty string in between proper interpolations";
+
+    // NULL terminator requires special care
+    expected = "*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$3\r\nb\0r\r\n";
+    std::string result = formatCommand("SET %b %b","foo",(size_t)3,"b\0r",(size_t)3);
+    bool cmp = memcmp(expected, result.data(), 4+4+(3+2)+4+(3+2)+4+(3+2));
+    ASSERT_TRUE(cmp == 0) << "%%b string interpolation";
+
+    expected = "*3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$0\r\n\r\n";
+    ASSERT_EQ(expected, formatCommand("SET %b %b","foo",(size_t)3,"",(size_t)0))
+        << "%%b and an empty string";
+
+    expected =  "*3\r\n$3\r\nSET\r\n$1\r\n%\r\n$1\r\n%\r\n";
+    ASSERT_EQ(expected, formatCommand("SET %% %%"))
+        << "literal %%";
+}
+
+TEST_F(FormatterTest, testTypeWidth) {
+    char *cmd;
+    int len;    
+
+    INTEGER_WIDTH_TEST("d", int);
```

Comment:
It would be better if each test would have a different function, e.g.

```
TEST_F(FormatterTest, testIntWidthChar)...
```

Or, if you want to get really fancy, I know that GoogleTest has a combinatorial test functionality, see `TEST_P`